NitroxPatcher/Patches/BuilderTool_Patch.cs

```diff
@@ -4,19 +4,69 @@
 using System.Reflection.Emit;
 using Harmony;
 using NitroxClient.GameLogic;
+using NitroxModel.Core;
 using NitroxModel.Helper;
 using UnityEngine;
 
 namespace NitroxPatcher.Patches
 {
+    public class BuilderTool_OnHover_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(BuilderTool);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("OnHover", BindingFlags.NonPublic | BindingFlags.Instance,
+                                                                                 null, new[] { typeof(Constructable) }, null);
+
+        public static bool Prefix(BuilderTool __instance, bool __result, Constructable constructable)
+        {
+            bool _result = true;
+
+            _result = NitroxServiceLocator.LocateService<Crafting>().BuilderTool_Pre_OnHover(__instance.gameObject, constructable);
+            if (_result == false)
+            {
+                return _result;
+            }
+            //Add more checks here later for other Objects
+
+            return _result;
+        }
+
+        public static void Postfix(BuilderTool __instance, Constructable constructable)
+        {
+            if (constructable.techType == TechType.Fabricator || constructable.techType == TechType.Workbench)
+            {
+                NitroxServiceLocator.LocateService<Crafting>().BuilderTool_Post_OnHover(__instance.gameObject, constructable);
+            }
+        }
+
+        public override void Patch(HarmonyInstance harmony)
+        {
+            PatchPostfix(harmony, TARGET_METHOD);
+        }
+    }
+
+
     public class BuilderTool_HandleInput_Patch : NitroxPatch
     {
         public static readonly Type TARGET_CLASS = typeof(BuilderTool);
         public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("HandleInput", BindingFlags.NonPublic | BindingFlags.Instance);
 
         public static readonly OpCode INJECTION_OPCODE = OpCodes.Callvirt;
-        public static readonly object INJECTION_OPERAND = typeof(Constructable).GetMethod("SetState", BindingFlags.Public | BindingFlags.Instance);
-        
+        public static readonly object INJECTION_OPERAND = typeof(Constructable).GetMethod("SetState", BindingFlags.Public | BindingFlags.Instance);
+
+        public static bool Prefix(BuilderTool __instance)
+        {
+            bool _result = true;
+
+            _result = NitroxServiceLocator.LocateService<Crafting>().BuilderTool_Pre_HandleInput(__instance.gameObject);

```

Comment:
As stated above, we should give all of these good, descriptive names.  From here it is not clear what the result entails.